### ISSUE TITLE
Regression between CVC4 and CVC5

### ISSUE THREAD

**Issue by rod-chapman** [cvc5/cvc5#9891]
The attached smt2 file shows a proof regression from cvc4 to cvc5. The proof arises from a simple integer inequality for proof of an assertion in a SPARK/Ada program.

I am using CVC4 1.8 and CVC5 1.0.5, both on macOS.

$cvc4 --tlimit=10000 issue9890.smt2
unsat

while

$cvc5 --tlimit=10000 issue9890.smt2
cvc5 interrupted by timeout.
zsh: abort      cvc5 --tlimit=10000 sm3.adb_458_28_assert_1.smt2

[issue9890.smt2.gz](https://github.com/cvc5/cvc5/files/12029794/issue9890.smt2.gz)

**Comment by ajreynol** [comment]
Note: it seems that CVC4 is solving this quickly due to its BV inequality solver, which is not available in cvc5.  The conflict CVC4 1.8 emits just before answering "unsat" has this form:
```
(and 
(not (bvult #b1000000000000000000000000000000000000000000000000000000000000000 x)) 
(not (bvult x #b0000000000000000000000000000000000000000000000000000000000000001)) 
(bvult (bvadd #b1000000000000000000000000000000000000000000000000000000000000000 (bvneg y)) (bvadd x (bvneg y))) 
(not (bvult x y)))
```
See `cvc4 --dump=t-lemmas`.

**Comment by rod-chapman** [comment]
Is there a plan to replace this technology in CVC5 with something equivalent or similar?

**Comment by mpreiner** [comment]
Yes, we have plans to implement improved versions of these techniques in cvc5.
